src/app/src/App.js

```diff
@@ -8,13 +8,37 @@ import Header from './Header';
 import Footer from './Footer';
 import SensorOverview from './SensorOverview';
 
+import { POLLING_INTERVAL } from './constants';
+import SENSORS from './sensors';
+
+import { pollSensor } from './sensor.actions';
+
+
 class App extends Component {
+    componentDidMount() {
+        // Poll for new sensor data immediately and on a timed cycle thereafter
```

Comment:
It appears that two requests are being fire immediately for each sensor. I believe that is because of the function invocation on line 28. To achieve the desired behavior it may work to extract the `fetchSensorData` function into a stand alone function, call it directly here, and then setup the `setInterval`



![image](https://user-images.githubusercontent.com/1042475/53105727-a8836900-34ff-11e9-8107-fa36cf2c2be1.png)

Reply:
So when I step through this with a debugger, any one function is only called 3 times. The duplicate call in the log shows an empty diff. 🤔

Reply:
Oh wow it's because action calls from another action were wrapped in `dispatch`